### ISSUE TITLE
Backward compatibility break after upgrading to 2.3.1 - link labels

### ISSUE THREAD

**Issue by dorsha** [clientIO/joint#882]
I got the following two backward compatibility issues after upgrading to 2.3.1:
1. Label text color has changed (see below images)
2. In 2.2.0, I was able to click the link label and get the `cell:pointerup` event, not I'm not, and it breaks my flow

2.2.0

![image](https://user-images.githubusercontent.com/5429033/40127708-b1a9f1f0-5938-11e8-81bc-19220a952754.png)

2.3.x

![image](https://user-images.githubusercontent.com/5429033/40127736-c36b72ec-5938-11e8-8d3a-b78ce53f77cb.png)

In 2.2.0 the `cell:pointerup` called when clicking the link label, and now it doesn't work anymore.

![image](https://user-images.githubusercontent.com/5429033/40127904-1480ad82-5939-11e8-831f-9e2f5d7034ac.png)

Please suggest what should I do for the above two issues.

Thanks

**Comment by kumilingus** [comment]
Hi @dorsha, were you able to fix the label color?

**Comment by dorsha** [comment]
Not yet, we cruelty staying with version 2.2.0.

**Comment by kumilingus** [comment]
We're about to release a patch. @dorsha, any chance sending me some code so I can reproduce the label text issue? (roman@client.io). Thank You.

**Comment by dorsha** [comment]
Yep, below you can see the labels part, there is no fill for the text, I guess I can provide a fill.. but the question is what has changed that cause the text color to be black?

![image](https://user-images.githubusercontent.com/5429033/40841252-7f0cdf24-65b2-11e8-8a59-605dbfe345ac.png)

**Comment by kumilingus** [comment]
Right. How did you add the label to the link? Could you paste here the JavaScript code? e.g.
```javascript
link.label(0, { attrs: { text: { text: 'yes' }}});
```
Did you change the label markup?

**Comment by dorsha** [comment]
I only set the fill attribute, did not touch the text attribute.
Here is the code, I hope you will understand the flow:

joint.dia.Paper creation:
![image](https://user-images.githubusercontent.com/5429033/41052785-77f5ae1e-69c2-11e8-8b9d-3e8cfa61edfa.png)

![image](https://user-images.githubusercontent.com/5429033/41052806-827cfa5e-69c2-11e8-9eb4-c3d53b5c0d70.png)

**Comment by kumilingus** [comment]
Hi @dorsha, the issue was introduced in JointJS `v2.0.1` with this [PR](https://github.com/clientIO/joint/pull/745). It's fixing another issue in IE. Previously the fill color of the `SVGText` was inherited from its parent `SVGGroup` (`".label"`). This has changed as the fill color is now set directly on the `SVGText`. Could you set the desired fill on `SVGText`?

**Comment by dorsha** [comment]
Yes, I will set the color myself.

**Comment by dorsha** [comment]
What about the label click issue?

**Comment by kumilingus** [comment]
Did the solution I mentioned earlier (https://github.com/clientIO/joint/issues/882#issuecomment-389819530) work for you? I've already commited the changes. These will become part of the patch release.

**Comment by dorsha** [comment]
I tried, but had a problem, so could not test it. Will wait for the patch.

**Comment by orius123** [comment]
@kumilingus hi, I tried your fix but it doesn't support `cell/link:pointreup` event. any suggestion how to add support for this as well?
when will the patch be released?

**Comment by orius123** [comment]
also in `cell:pointermove` event argument I don't longer have `_dx` and `_dy` any way of getting them? 
We use it for our labels to move only on the link.
```js
paper.on('cell:pointermove', (event) => {
    // this code handle label movements on the link
    if (event.targetView && paper.options.interactive && linkHasLabels(event)) {
      const clickPoint = { x: event._dx, y: event._dy };
      const lengthTotal = event.sourcePoint.manhattanDistance(event.targetPoint);
      const length = event.sourcePoint.manhattanDistance(clickPoint);
      let position = round(length / lengthTotal, 2);
      if (position <= 0.9 && position >= 0.1) {
        position = clamp(round(length / lengthTotal, 2), 0.1, 0.9);
        event.model.label(0, { position });
      }
    }
  });
```

**Comment by kumilingus** [comment]
Hi @orius123,  both `cell:pointerup` and `link:pointerup` works for me (they are triggered, when I stop dragging a label. What scenario does not work for you?

We are planning to release the new version this week.

For the deltas, these private properties have been removed from the CellVIew. You have 2 options.
1. Keep track on deltas yourself or use `x` and `y` from the handlers arguments.
```javascript
// a
paper.on({
  'cell:pointerdown': (cellView, event, x, y) => {
     event.data = { x, y };
   },
  'cell:pointermove': (cellView, event, x, y) => {
    console.log(x - event.data.x, y - event.data.y);
    event.data.x = x;
    event.data.y = y;
  }
});
// b
paper.on('link:pointermove', (linkView, event, x, y) => { /* use x and y */ });
```
2. Don't use deltas at all :) 
```javascript
// reseting offset
paper.on({ 'link:pointermove': (linkView) => {
  linkView.model.label(0, { position: { offset: null }});
}});
```

**Comment by kumilingus** [comment]
Fixed in `v2.1.3`.
